agent/requestor/src/market.rs

```diff
@@ -28,29 +32,31 @@ pub(crate) fn build_demand(
         },
     });
 
+    let common_constraint = Constraints::new_clause(
+        And,
+        vec![
+            ConstraintKey::new("golem.inf.mem.gib").greater_than(ConstraintValue::new(0.5)),
+            ConstraintKey::new("golem.inf.storage.gib").greater_than(ConstraintValue::new(1)),
+            ConstraintKey::new("golem.com.pricing.model").equal_to(ConstraintValue::new("linear")),
+        ],
+    );
     let subnet_constraint = match subnet {
         Some(subnet) => {
             log::info!("Using subnet: {}", subnet);
             properties.as_object_mut().unwrap().insert(
                 "golem.node.debug.subnet".to_string(),
                 serde_json::Value::String(subnet.clone()),
             );
-            format!("(golem.node.debug.subnet={})", subnet.clone())
+            common_constraint.and(Constraints::new_single(
+                ConstraintKey::new("golem.node.debug.subnet").equal_to(ConstraintValue::new(serde_json::Value::String(subnet.clone())))
```

Comment:
Why do you create ConstraintValue from serde_json::Value::String? Does it work?
Note that when we add subnet to properties (line 48), we must wrap it in serde_json::Value, because `properties` is serde structure. Here subnet can be used directly.

Reply:
```
   Compiling ya-requestor v0.1.0 (/home/dariusz/repo/yagna/agent/requestor)
error[E0308]: mismatched types
  --> agent/requestor/src/market.rs:41:13
   |
41 |             subnet.clone(),
   |             ^^^^^^^^^^^^^^
   |             |
   |             expected enum `serde_json::value::Value`, found struct `std::string::String`
   |             help: try using a variant of the expected enum: `serde_json::value::Value::String(subnet.clone())`
```

Reply:
Hmm, maybe `ConstraintValue::from(subnet.clone())` could help?
New should work for all types that implement `Into<serde_json::Value>`, so I would expect it to work even with new. @filipgolem

Reply:
Does `cnts = cnts.and(constraints!["golem.node.debug.subnet" == subnet.clone(),]);` work?

Reply:
Yes, it does. Thanks